src/components/tables/DataTable/table/DataTable.tsx

```diff
@@ -18,6 +19,115 @@ import type { DataTableStatus } from '../DataTableContext.tsx';
 
 import cl from './DataTable.module.scss';
 
+const getFlexValueFromColumn = <D extends object>(col: ReactTable.ColumnInstance<D>) => {
+  return (col?.bkColumnWidth?.flex ?? 1); // defaults to 1
+};
+
+const getFlexColumnWidth = <D extends object>(col: ReactTable.ColumnInstance<D>) => {
+  return (col?.bkColumnWidth?.width ?? 0);
+};
+
+const isFlexColumn = <D extends object>(col: ReactTable.ColumnInstance<D>) => {
+  return (col && Object.hasOwn(col, 'bkColumnWidth'));
+};
+
+const requireNewCol = <D extends object>(col: ReactTable.ColumnInstance<D>) => {
+  return (isFlexColumn(col) && (col.bkColumnWidth?.flex ?? 0) > 0);
+};
+
+const calculateFlexPercentage = <D extends object>(
+  allColumns: ReactTable.ColumnInstance<D>[],
+  targetColumn: ReactTable.ColumnInstance<D>
+) => {
+  if (!isFlexColumn(targetColumn)) {
+    return 0;    
+  }
+
+  const targetColumnFlex = getFlexValueFromColumn(targetColumn);
+  const totalFlex = allColumns.reduce((total, column) => total + getFlexValueFromColumn(column), 0);
+  
+  if (targetColumnFlex === 0) {
+    return 0;
+  }
+  
+  return totalFlex > 0 ? Math.round((targetColumnFlex / totalFlex) * 100) : 0;
+};
+
+export const generateCustomColGroup = <D extends object>(
+  allColumns: ReactTable.ColumnInstance<D>[],
+  columnGroups?: React.ReactNode): React.ReactNode | null => {
+  if (columnGroups && React.isValidElement(columnGroups) && columnGroups.type === 'colgroup') {
```

Comment:
I think just `if (columnGroups)` here is sufficient, you don't need to check the element type.